03_letters/alanhillgemann/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, strings.Join(sortLetters(letters("aba")), "\n"))
+}
+
+func letters(s string) map[rune]int {
+	runeMap := make(map[rune]int)
+	runeSlice := []rune(s)
+	for i := 0; i <= len(runeSlice)-1; i++ {
```

Comment:
You can simplify the end-test: `i < len(runeSlice)`. However, you can go further and use range-loop to simplify and avoid the slice copy:
```go
for _, r := range s {
	runeMap[r]++
}
```
This works because Go's range-loop has a special case that iterates over the decoded runes of a string instead of its bytes. See note 2 under https://golang.org/ref/spec#For_range.

Reply:
Fixed